src/orders/components/OrderListDatagrid/OrderListDatagrid.tsx

```diff
@@ -0,0 +1,126 @@
+import ColumnPicker from "@dashboard/components/ColumnPicker";
+import Datagrid from "@dashboard/components/Datagrid/Datagrid";
+import { useColumnsDefault } from "@dashboard/components/Datagrid/hooks/useColumnsDefault";
+import {
+  DatagridChangeStateContext,
+  useDatagridChangeState,
+} from "@dashboard/components/Datagrid/hooks/useDatagridChange";
+import { TablePaginationWithContext } from "@dashboard/components/TablePagination";
+import { OrderListQuery } from "@dashboard/graphql";
+import { OrderListUrlSortField } from "@dashboard/orders/urls";
+import { ListProps, RelayToFlat, SortPage } from "@dashboard/types";
+import { Item } from "@glideapps/glide-data-grid";
+import { Box } from "@saleor/macaw-ui/next";
+import React, { useCallback } from "react";
+import { useIntl } from "react-intl";
+
+import { useColumns, useGetCellContent } from "./datagrid";
+import { messages } from "./messages";
+import { canBeSorted, getColumnMetadata, getOrdersRowsLength } from "./utils";
+
+interface OrderListDatagridProps
+  extends ListProps,
+    SortPage<OrderListUrlSortField> {
+  orders: RelayToFlat<OrderListQuery["orders"]>;
+  onRowClick: (id: string) => void;
+}
+
+export const OrderListDatagrid: React.FC<OrderListDatagridProps> = ({
+  orders,
+  disabled,
+  settings,
+  onUpdateListSettings,
+  onSort,
+  sort,
+  onRowClick,
+}) => {
+  const intl = useIntl();
+  const datagrid = useDatagridChangeState();
+  const availableColumns = useColumns(sort);
+  const ordersLength = getOrdersRowsLength(orders, disabled);
+
+  const {
+    availableColumnsChoices,
+    columnChoices,
+    columns,
+    defaultColumns,
+    onColumnMoved,
+    onColumnResize,
+    onColumnsChange,
+    picker,
+  } = useColumnsDefault(availableColumns);
+
+  const handleHeaderClick = useCallback(
+    (col: number) => {
+      const { columnName, columnId } = getColumnMetadata(columns[col].id);
+
+      if (canBeSorted(columnName)) {
+        onSort(columnName, columnId);
+      }
+    },
+    [columns, onSort],
+  );
+
+  const handleRowClick = useCallback(
+    ([_, row]: Item) => {
+      const rowData = orders[row];
+      onRowClick(rowData.id);
+    },
+    [onRowClick, orders],
+  );
```

Comment:
**suggestion:** These functions look similar to product datagrid ones, perhaps they could be extracted?